ui/lib/replication/addon/controllers/replication-mode.js

```diff
@@ -1,13 +1,26 @@
 import { alias } from '@ember/object/computed';
 import { inject as service } from '@ember/service';
 import Controller from '@ember/controller';
+import { task, timeout } from 'ember-concurrency';
 
 export default Controller.extend({
   rm: service('replication-mode'),
   replicationMode: alias('rm.mode'),
+  waitForNewClusterToInit: task(function*(replicationMode) {
+    // waiting for the newly enabled cluster to init
+    // this ensures we don't hit a capabilities-self error, called in the model of the mode/index route
+    yield timeout(1000);
```

Comment:
I don't love arbitrary timeouts, but I trust you that this was necessary. For diligence: is there a way to kick this off after the other thing has finished from the index route? Looks like it does the transition fine when the mode is `dr` ?

Reply:
It has everything to do with hitting the model on the route it's transitioning to.  For DR it's not transitioning to the same route, only in this specific case (Enabling a Performance Secondary) does it follow this flow: `index ->transitionTo-> mode/index`. Because this is where that `transitionTo` occurs, this is the only place I was able to interrupt it before hitting the routes' model that calls the capabilities-self endpoint.  Essentially, right here is the last step before we transition.  There's nothing - at least that I could tell - that we could wait for to complete.  We don't appear to know what's happening with the cluster's initialization at this point.  🤷‍♀️

Reply:
i was thinking about this too -- an arbitrary timeout feels like it could have the potential to be flaky. i know you have tried many ideas, but have you considered moving the fetch for `canAddSecondary` and such into the controller? i took a cursory glance at the ember guides for controllers and noticed:
> Controllers are used as an extension of the model loaded from the Route. Any attributes or actions that we want to share with components used on that Route could be defined on the Controller and passed down through the Route’s template.

maybe it's possible to get the model from the route and add the additional properties to it in the controller, since the model will exist by the time it gets here?

https://guides.emberjs.com/release/routing/controllers/#toc_where-and-when-to-use-controllers

i'm not sure, but some guidance from others would be helpful here too -- @meirish, @DingoEatingFuzz, or @johncowen, do any of you have advice for us on how to ensure a network request in the model doesn't happen until the model is already "ready"?

Reply:
Not having looked massively at this code yet, my question would be:

What's the 'signal' so you know when this is 'ready'?

It sounds like there isn't one?

> There's nothing - at least that I could tell - that we could wait for to complete.

So a follow up question would be, whilst potentially horrible, is there something you could poll so you know when this is 'ready'?

Reply:
Hmm yeah that's tricky - I'd expect replication status or cluster status to tell you if it's bootstrapping (or some other status as it's changing). Alternatively does _every_ request return a 40x error? If so you could poll until you don't get 40x's (not great, but more of signal like John mentioned). Or even poll the capabilities-self endpoint and catch the errors until it doesn't return one.